base/linalg/lapack.jl

```diff
@@ -664,7 +675,7 @@ for (gesvx, elty) in
             ferr      = similar(A, $elty, nrhs)
             berr      = similar(A, $elty, nrhs)
             work      = Array($elty, 4n)
-            iwork     = Array($elty, n)
+            iwork     = Array($elty, 2n)
```

Comment:
From the above I can't tell if I've done something wrong, LAPACK is wrong, or the GC is doing something bad. I can try to reproduce this with a C test case over the next few days. In the interim, should we hold off on merging this or keep the `4n` instead of `2n` for now?